src/state/slices/swappersSlice/swappersSlice.ts

```diff
@@ -102,7 +106,45 @@ export const swappersApi = createApi({
         return { data: maybeQuote }
       },
     }),
+    getThorTradeQuote: build.query<
+      Result<LifiTradeQuote<false>, SwapErrorRight>,
+      GetTradeQuoteInput
+    >({
+      queryFn: async (getTradeQuoteInput: GetTradeQuoteInput, { getState }) => {
+        const state: ReduxState = getState() as ReduxState
+        // selectCryptoMarketData is used because prives are always in USD
+        const marketDataById = selectCryptoMarketData(state) as MarketDataById<AssetId>
+        const feeAsset = selectFeeAssetById(state, getTradeQuoteInput.sellAsset.assetId) as
+          | Asset
+          | undefined
```

Comment:
If we want to fix the need to use type assertions, upstream:

```ts
export const selectFeeAssetById = createCachedSelector(
  selectAssets,
  (_state: ReduxState, assetId: AssetId) => assetId,
  (assetsById, assetId): Asset | undefined => getFeeAssetByAssetId(assetsById, assetId),
)((_s: ReduxState, assetId: AssetId) => assetId ?? 'assetId') as (
  state: ReduxState,
  assetId: AssetId,
) => ReturnType<typeof getFeeAssetByAssetId>
```

Reply:
absolute legend